test/gtest/uct/test_uct_query.cc

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+extern "C" {
```

Comment:
@yosefe can we change the file name to `test_uct_v2_query` or move this to `uct/v2/` folder ?

Reply:
i think we actually want to transition tests to use the new private APIs, and test old APIs in something like "test_uct_compat.cc"

Reply:
well, then it makes even more sense to add some prefix for now. Once the API stable we can move  uct_*.cc -> uct_*_compat.cc and uct_*_compat.cc -> uct_*.cc

Reply:
Let's leave it `test_uct_query.cc`, as discussed with @yosefe .

Reply:
this way, after we decide to switch to new API, will not need to change tests

Reply:
@yosefe and I discussed offline that it makes sense as for now to move the tests to `uct/v2`. Later we can rearrange this

Reply:
+1
@shamisp can we work on the _v2 renaming macros in another PR? this PR does not change an existing function so there is no name collision

Reply:
@yosefe yes, but I was commenting on the test location (aka move it to v2 subfolder). I agree with _v2 C macro development in a separate PR.

Reply:
@shamisp +1 for the test location